pilot/pkg/networking/core/cluster_builder.go

```diff
@@ -279,6 +280,23 @@ func (cb *ClusterBuilder) applyMetadataExchange(c *cluster.Cluster) {
 	}
 }
 
+// If there is a "." in the cluster name (besides .svc.cluster.local), it will be replaced with "_"
+// in order to prevent broken telemetry
+func sanitizeName(c *cluster.Cluster) {
+	// We have a very specific regex in envoy_bootstrap.json that matches this suffix
+	// specifically. If we find it in the cluster name, we won't add an alt stat name.
+	prefix, found := strings.CutSuffix(c.Name, ".svc.cluster.local")
```

Comment:
what about a cluster with custom trust domain?

Reply:
The regex for cluster_name doesn't take that into account so those are broken today under the status quo

Reply:
I'm not sure why we need to; this substitution is solely based on the regex in envoy_bootstrap. That regex hardcodes cluster.local so we do that here as well.